unlock-app/src/components/content/EventContent.tsx

```diff
@@ -0,0 +1,201 @@
+import React from 'react'
+import Head from 'next/head'
+import { Button, Tooltip } from '@unlock-protocol/ui'
+import { pageTitle } from '../../constants'
+import { useRouter } from 'next/router'
+import { AppLayout } from '../interface/layouts/AppLayout'
+import { useMetadata } from '~/hooks/metadata'
+import LoadingIcon from '../interface/Loading'
+import { AddressLink } from '../interface/AddressLink'
+import { ReactMarkdown } from 'react-markdown/lib/react-markdown'
+import { FaCalendar, FaRegCalendarPlus, FaClock } from 'react-icons/fa'
+import { GoLocation } from 'react-icons/go'
+import { FiTwitter } from 'react-icons/fi'
+import Link from 'next/link'
+import { useAuth } from '~/contexts/AuthenticationContext'
+
+const formatEventData = (metadata: any) => {
+  const accentColor = metadata.background_color
+    ? `#${metadata.background_color}`
+    : 'bg-ui-secondary-200'
+
+  let date, time, address
+
+  metadata.attributes.forEach(({ trait_type, value }) => {
+    if (trait_type === 'event_start_date') {
+      const options = {
+        weekday: 'long',
+        year: 'numeric',
+        month: 'long',
+        day: 'numeric',
+      }
+
+      date = new Date(Date.parse(value)).toLocaleDateString(undefined, options)
+    } else if (trait_type === 'event_start_time') {
+      time = value
+    } else if (trait_type === 'event_address') {
+      address = value
+    }
+  })
+
+  return {
+    title: metadata.name,
+    description: metadata.description, // in Markdown!
+    image: metadata.image,
+    date,
+    time,
+    address,
+    accentColor,
+  }
+}
+
+export const EventContent = () => {
+  const router = useRouter()
+  if (!router.query) {
+    return <LoadingIcon></LoadingIcon>
+  }
+
+  return (
+    <AppLayout showLinks={false} authRequired={false} title="">
+      <Head>
+        <title>{pageTitle('Event')}</title>
+      </Head>
+      <div className="md:w-3/4 m-auto">
+        {(!router.query.lockAddress || !router.query.network) && (
```

Comment:
More readable to extract network and lock address at the top.

Reply:
good point

Reply:
Done!